src/functions/layerGroup.js

```diff
@@ -0,0 +1,32 @@
+import { provide } from "vue";
+import { props as layerProps, setup as layerSetup } from "./layer";
+
+export const props = {
+  ...layerProps,
+};
+
+export const setup = (props, leafletRef) => {
+  const { options: layerOptions, methods: layerMethods } = layerSetup(
+    props,
+    leafletRef
+  );
+
+  const options = {
+    ...layerOptions,
+  };
+
+  const methods = {
+    ...layerMethods,
+    addLayer(layer) {
+      leafletRef.value.addLayer(layer.leafletObject);
+    },
+    removeLayer(layer) {
+      leafletRef.value.removeLayer(layer.leafletObject);
+    },
+  };
+
+  provide("addLayer", methods.addLayer);
+  provide("removeLayer", methods.removeLayer);
```

Comment:
I'm more than happy to leave this as-is for now, especially since this isn't the first component with this here. But reviewing this pattern has raised a question for me just now, so I thought I'd mention it while I was thinking about it...

Should we be putting all of our `provide` calls in the `.vue` component files themselves, instead of these helper functions? I think that it conceptually makes more sense to keep Vue-specific behaviour in the Vue files. For one thing, it would mean that these `.js` files would be truly vanilla and could be tested independently of Vue. For another, things provided by a component are essentially outputs of that component, so it might make navigating the code clearer if those outputs were easily visible right in the component itself. Finally, I can also imagine that having `provide` statements in the `.vue` files might make it easier for documentation extractors to recognize and document those provided outputs.

Reply:
@mikeu I was thinking the same, but doing so we would break the `matrioska` structure that mimics leaflet classes. Maybe we should actually do it, but I think we should discuss this in its dedicated issue! I will open one!